File: composeApp/src/androidMain/kotlin/com/sircjarr/marvelrivalsherolookup/ui/res/RateColor.kt
package com.sircjarr.marvelrivalsherolookup.ui.res

import androidx.compose.ui.graphics.Color

val Number.winRateColor: Color get() {
    return when(this.toFloat()) {
        in 50f..100f -> ColorRes.green
        in 30f..50f -> ColorRes.yellow
        else -> ColorRes.red
    }
}

val Number.pickRateColor: Color get() {
    return when(this.toFloat()) {
        in 12f..100f -> ColorRes.green
        in 5f..12f -> ColorRes.yellow
        else -> ColorRes.red
    }
}

File: iosApp/iosApp/ui/res/RateColor.swift
import SwiftUI

extension Double {
    var winRateColor: Color {
        return switch self {
        case 50...100: Color.green
        case 30...50: Color.yellow
            default: Color.red
        }
    }
    
    var pickRateColor: Color {
        return switch self {
        case 12...100: Color.green
        case 5...12: Color.yellow
            default: Color.red
        }
    }
}
